examples/svelte/src/App.svelte

```diff
@@ -0,0 +1,66 @@
+<script>
+    import { router } from './router.js';
+    import { uuid } from './utils.js';
+
+    import Header from './Header.svelte';
+    import Footer from './Footer.svelte';
+    import Item from './Item.svelte';
+
+    import "./app.css";
+    import "todomvc-app-css/index.css";
+    import "todomvc-common/base.css";
+
+    let currentFilter = "all";
+    let items = [];
+    let editing = null;
+
+    function addItem(event) {
+        items.push({
+            id: uuid(),
+            description: event.detail.text,
+            completed: false,
+        });
+        items = items;
+    }
+
+    function removeItem(index) {
+        items.splice(index, 1);
+        items = items;
+    }
+
+    function toggleAllItems(event) {
+        const checked = event.target.checked;
+        items = items.map((item) => ({
+        ...item,
+        completed: checked,
+        }));
+    }
+
+    function removeCompletedItems() {
+        items = items.filter((item) => !item.completed);
+    }
+    
+    router(route => currentFilter = route).init();
+
+    $: filtered = currentFilter === "all" ? items : currentFilter === "completed" ? items.filter((item) => item.completed) : items.filter((item) => !item.completed);
+    $: numActive = items.filter((item) => !item.completed).length;
+    $: numCompleted = items.filter((item) => item.completed).length;
+</script>
+
+<Header on:addItem={addItem} />
+
+{#if items.length > 0}
+    <main class="main">
+        <div class="toggle-all-container">
+            <input id="toggle-all" class="toggle-all" type="checkbox" on:change={toggleAllItems} checked={numCompleted === items.length} />
+            <label for="toggle-all">Mark all as complete</label>
+        </div>
+        <ul class="todo-list">
+            {#each filtered as item, index (item.id)}
+                <Item bind:item editing={editing} on:removeItem={() => removeItem(index)} />
```

Comment:
In the original implementation `editing` is the index of the todo that's currently being edited. Here, it's a per-component piece of state, and it looks like we're relying on the `blur` event to ensure that it's only true for one item at a time. I think we can therefore remove it

```suggestion
                <Item bind:item on:removeItem={() => removeItem(index)} />
```

Reply:
this is also used to assign a [conditional class](https://github.com/tastejs/todomvc/blob/5d8525402426aedce9448c2fa7496b224e4c0a36/examples/svelte/src/Item.svelte#L41) for styling and a [conditional render](https://github.com/tastejs/todomvc/blob/5d8525402426aedce9448c2fa7496b224e4c0a36/examples/svelte/src/Item.svelte#L49).

Reply:
Yes, but it's state that's internal to `Item.svelte`. The prop here isn't doing anything (it never changes in `App.svelte`) and can be removed